lib/wrapThenable.js

```diff
@@ -10,6 +10,16 @@ function wrapThenable (thenable, reply) {
     if (reply[kReplyHijacked] === true) {
       return
     }
+    // since wrap thanable will always be executed in async handler, when
+    // header is already sent or request is aborted. it means either we
+    // sent already or we can not send anymore response to the user.
+    //
+    // reply.sent is not enough to check in this case, because the response
+    // may not be ended properly and the writableEnded will be false in this
+    // case.
```

Comment:
Maybe we should amend `reply.sent` so that it's true in this case. I think it would match the intent.

Reply:
We should not change the `reply.sent` because it will change too much of the current behavior and may result in breaking change.

Reply:
I updated the `reply.sent` but I still bring a extra symbol.
The reason behind is that `reply.sent` has it's own meaning. It indicate whether we call the `reply.send` already.

But what we need is whether the `reply` is writable or not.

Reply:
I can remove that extra symbol because it is used in one place only.
Only the `async` handler may fall in this strange condition.

Reply:
Updated

Reply:
I decided not to update `reply.sent` since it breaks `reply.test.js`.